src/mapper/pkg/resourcevisibility/svc_reconciler.go

```diff
@@ -1,36 +1,53 @@
 package resourcevisibility
 
 import (
+	"bytes"
 	"context"
+	"fmt"
+	"github.com/hashicorp/golang-lru/v2/expirable"
 	"github.com/otterize/intents-operator/src/shared/errors"
 	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
 	"github.com/otterize/network-mapper/src/mapper/pkg/cloudclient"
+	"github.com/otterize/network-mapper/src/mapper/pkg/config"
 	"github.com/otterize/network-mapper/src/mapper/pkg/graph/model"
 	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"hash/crc32"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"slices"
 	"time"
 )
 
 type ServiceReconciler struct {
 	client.Client
 	injectablerecorder.InjectableRecorder
-	otterizeCloud cloudclient.CloudClient
-	kubeFinder    KubeFinder
+	otterizeCloud                    cloudclient.CloudClient
+	kubeFinder                       KubeFinder
+	namespaceToReportedServicesCache *expirable.LRU[string, []byte]
 }
 
 type KubeFinder interface {
 	ResolveOtterizeIdentityForService(ctx context.Context, service *corev1.Service, now time.Time) (model.OtterizeServiceIdentity, bool, error)
 }
 
+func OnEvict(key string, _ []byte) {
+	logrus.Debugf("Namespace %s evicted from cache, you may change configuration to increase cache size or TTL", key)
```

Comment:
Use `logrus.WithField("namespace", key).Debug(...)` when the log message is parameterized by something with very strict format. It makes the logs nicer.